tests/rancher-istio/rancher_istio.go

```diff
@@ -0,0 +1,947 @@
+					pilotEnabled := checker.MustRenderValue[bool](tc, ".Values.pilot.enabled")
+
+					if pdb.Name != "pilot" {
+						return
+					}
+
+					if !pilotEnabled {
+						return
+					}
+
+					pilotPodDisruptionBudget, _ := checker.RenderValue[string](tc, ".Values.pilot.podDisruptionBudget")
+					assert.Equal(tc.T, pilotPodDisruptionBudget, pdb.Spec)
+				}),
+			},
+		},
+
+		{
+			Name: "Check .Values.egressGateways.type",
+			Covers: []string{
+				".Values.egressGateways.enabled",
+				".Values.egressGateways.type",
+			},
+
+			Checks: test.Checks{
+				checker.PerWorkload(func(tc *checker.TestContext, obj metav1.Object, podTemplateSpec corev1.PodTemplateSpec) {
+					egressGatewaysEnabled := checker.MustRenderValue[bool](tc, ".Values.egressGateways.enabled")
+
+					if !egressGatewaysEnabled {
+						return
+					}
+
+					egressGatewaysType, _ := checker.RenderValue[string](tc, ".Values.egressGateways.type")
+
+					if egressGatewaysType != "NodePort" {
+						return
+					}
+				}),
+			},
+		},
+
+		{
+			Name: "Check .Values.egressGateways.hpaSpec",
+			Covers: []string{
+				".Values.egressGateways.enabled",
+				".Values.egressGateways.hpaSpec",
+			},
+
+			Checks: test.Checks{
+				checker.PerResource(func(tc *checker.TestContext, hpa *autoscalingv2.HorizontalPodAutoscaler) {
+					egressGatewaysEnabled := checker.MustRenderValue[bool](tc, ".Values.egressGateways.enabled")
+
+					if !egressGatewaysEnabled {
+						return
+					}
+
+					egressGatewaysHpaSpec, _ := checker.RenderValue[string](tc, ".Values.egressGateways.hpaSpec")
+					assert.Equal(tc.T, egressGatewaysHpaSpec, hpa.Spec)
+				}),
+			},
+		},
+
+		{
+			Name: "Check .Values.egressGateways.podDisruptionBudget",
+			Covers: []string{
+				".Values.egressGateways.enabled",
+				".Values.egressGateways.podDisruptionBudget",
+			},
+
+			Checks: test.Checks{
+				checker.PerResource(func(tc *checker.TestContext, pdb *policyv1beta1.PodDisruptionBudget) {
+					egressGatewaysEnabled := checker.MustRenderValue[bool](tc, ".Values.egressGateways.enabled")
+
+					if !egressGatewaysEnabled {
+						return
+					}
+
+					egressGatewaysPodDisruptionBudget, _ := checker.RenderValue[string](tc, ".Values.egressGateways.podDisruptionBudget")
+					assert.Equal(tc.T, egressGatewaysPodDisruptionBudget, pdb.Spec)
+				}),
+			},
+		},
+
+		{
+			Name: "Check .Values.ingressGateways.type",
+			Covers: []string{
+				".Values.ingressGateways.enabled",
+				".Values.ingressGateways.type",
+			},
+
+			Checks: test.Checks{
+				checker.PerWorkload(func(tc *checker.TestContext, obj metav1.Object, podTemplateSpec corev1.PodTemplateSpec) {
+					ingressGatewaysEnabled := checker.MustRenderValue[bool](tc, ".Values.ingressGateways.enabled")
+
+					if !ingressGatewaysEnabled {
+						return
+					}
+
+					ingressGatewaysType, _ := checker.RenderValue[string](tc, ".Values.ingressGateways.type")
+
+					if ingressGatewaysType != "NodePort" {
```

Comment:
This test doesn't do anything. Isn't there a specific type you need to assert for?

Reply:
updated